gap/projective/almostsimple.gi

```diff
@@ -1429,3 +1430,135 @@ FindHomMethodsProjective.SporadicsByOrders := function(ri,G)
+        maximalOrdersOverset := [18, 22, 24, 25, 28, 30, 31, 33, 37, 40, 42, 67],
+        maximalOrdersSubset := [31, 67]),
+    rec(name := "J4",
+        maximalOrdersOverset := [16, 23, 24, 28, 29, 30, 31, 33, 35, 37, 40, 42, 43, 44, 66],
+        maximalOrdersSubset := [37, 43]),
+    rec(name := "Fi23",
+        maximalOrdersOverset := [13, 14, 16, 17, 22, 23, 24, 26, 27, 28, 35, 36, 39, 42, 60],
+        maximalOrdersSubset := [17, 23]),
+    rec(name := "Fi24'",
+        maximalOrdersOverset := [16, 17, 22, 23, 24, 26, 27, 28, 29, 33, 35, 36, 39, 42, 45, 60],
+        maximalOrdersSubset := [17, 29]),
+    rec(name := "Th'",
+        maximalOrdersOverset := [19, 20, 21, 24, 27, 28, 30, 31, 36, 39],
+        maximalOrdersSubset := [19, 31, 39]),
+]);
+#! @BeginChunk NameSporadic
+#! This method returns a list of sporadic simple groups that <A>ri</A>
+#! could be. It does not recognise the Monster and the Baby Monster group. It
+#! is based on the Magma v2.24.10 function <C>RecognizeSporadic</C>.
+#!
+#! According to communication with Eamonn O'Brien
+# TODO G is unused
+FindHomMethodsProjective.NameSporadic := function(ri, G)
+    local orders, setOfOrders, maximalOrders, isMaximal,
+        namesOfPossibleSporadics, res, i, j, data, name;
+    orders := [];
+    #do we need SetPseudoRandomStamp?
+    for i in [1 .. 500] do
+        Add(orders, RandomElmOrd(ri, "NameSporadic", false).order);
+    od;
+    # Compute maximal orders. Maximal in the sense that it does not divide the
+    # order of another group element.
+    setOfOrders := AsSet(orders);
+    maximalOrders := [];
+    for i in [1..Length(setOfOrders)] do
+        isMaximal := true;
+        for j in [i+1..Length(setOfOrders)] do
+            if setOfOrders[j] mod setOfOrders[i] = 0 then
+                isMaximal := false;
+                break;
+            fi;
+        od;
+        if isMaximal then
+            Add(maximalOrders, setOfOrders[i]);
+        fi;
+    od;
+    namesOfPossibleSporadics := [];
+    for data in RECOG.NameSporadicData do
+        if IsSubset(maximalOrders, data.maximalOrdersSubset)
+                and IsSubset(data.maximalOrdersOverset, maximalOrders) then
+            Add(namesOfPossibleSporadics, data.name);
+        fi;
+    od;
+    # HN works a bit differently
+    if IsSubset([9, 12, 14, 19, 21, 22, 25, 30, 35, 40], maximalOrders)
+        and (IsSubset(maximalOrders, [19]) or IsSubset(maximalOrders, [21]))
+    then
+        Add(namesOfPossibleSporadics, "HN");
+    fi;
+    if ValueOption("DEBUGRECOGSPORADICS") <> fail then
+        return namesOfPossibleSporadics;
+    fi;
+    for name in namesOfPossibleSporadics do
+        Info(InfoRecog, 2, "Trying hint for ", name,
+             "...");
+        res := LookupHintForSimple(ri, Grp(ri), name);
+        if res = true then return Success; fi;
+        Info(InfoRecog, 2, "This did not work.");
+    od;
+    return false; # FIXME: false = NeverApplicable here really correct?
```

Comment:
Well, the only reason why it might succeed next time is because we missed some orders that excluded a potential sporadic... So in theory retrying might sometimes help, but only if we really didn't compute enough random elements. But doing this often would kill performance: if the group really is not sporadic then trying this again and again is useless.

Reply:
You're right. I'm changing this to `return NeverApplicable;`.